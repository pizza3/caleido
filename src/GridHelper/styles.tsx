import styled from 'styled-components'


export const Container = styled.div`
    position: absolute;
    width:100%;
    height: 100%;
    z-index:0;
    overflow:hidden;
`

export const HorizontalLine = styled.div`
    width:100%;
    position: absolute;
    height: 1px;
    background:#e7e7e7;
    top:${(props:{top:number})=>props.top+'px'};
`

export const VerticalLine = styled.div`
    width:1px;
    position: absolute;
    height: 100%;
    background:#e7e7e7;
    left:${(props:{top:number})=>props.top+'px'};
`

export const ReferencePoint = styled.div`
    width: 8px;
    position: absolute;
    border-radius: 50%;
    height: 8px;
    background:#e7e7e7;
    left:${(props:{left:number, top:number})=>props.left+'px'};
    top:${(props)=>props.top+'px'};
`

export const Diagonal = styled.div`
    width: 367.7px;
    height: 1px;
    background:#e7e7e7;
    position: absolute;
    transform-origin: left;
    transform: ${(props:{left:number, top:number, angle:number})=>`rotate(${props.angle}deg)`};
    left:${(props)=>props.left+'px'};
    top:${(props)=>props.top+'px'};

`

export const Diagonal2 = styled.div`
    width: 130px;
    height: 1px;
    background:#e7e7e7;
    position: absolute;
    transform-origin: left;
    transform: ${(props:{left:number, top:number, angle:number})=>`rotate(${props.angle}deg)`};
    left:${(props)=>props.left+'px'};
    top:${(props)=>props.top+'px'};

`

export const RotateLine =  styled.div`
    width:1px;
    position: absolute;
    height: 100%;
    background:#e7e7e7;
    bottom: 50%;
    left:${(props:{left:number,angle:number})=>props.left+'px'};
    transform-origin: bottom;
    transform: ${(props)=>`rotate(${props.angle}deg)`};
`
