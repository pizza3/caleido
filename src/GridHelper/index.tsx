import React, { Component } from 'react';
import { Container, VerticalLine, HorizontalLine, ReferencePoint, Diagonal, RotateLine, Diagonal2 } from './styles'
type Props = {
  mode: string
  sections: number
}

class Index extends Component<Props>{
  static defaultProps = {
    mode: 'Mirror',
    sections: 10
  }
  componentDidMount() {
    const This = this;
    window.addEventListener('resize',function(){
      This.forceUpdate();
    })
  }

  renderGrid = () => {
    const { mode } = this.props
    let grid: JSX.Element | JSX.Element[] | [] = [];
    switch (mode) {
      case 'Mirror':
        grid = this.mirrorGrid()
        break;
      case 'Rotation':
        grid = this.rotationalGrid()
        break;
      case 'Kaliedo':
        grid = this.rotationalGrid()
        break;
      case 'SquareRotation':
        grid = this.squareGrid()
        break;
      case 'SquareKaliedo':
        grid = this.squareGrid()
        break;
      case 'Hexagon':
        grid = this.hexagonGrid()
        break;
      case 'HexagonRotation':
        grid = this.triangleGrid()
        break;
      case 'HexagonKaliedo':
        grid = this.triangleGrid()
        break;
      default:
        grid = this.mirrorGrid()
        break;
    }
    return grid;
  }

  mirrorGrid = () => {
    const width = window.innerWidth - 250
    return (<VerticalLine top={width / 2} />)
  }

  rotationalGrid = () => {
    const width = window.innerWidth - 250
    const { sections } = this.props
    let secs = []
    for (let i = 0; i <= 360; i += 360 / sections) {
      secs.push(<RotateLine key={'RotateLine' + i} left={width / 2} angle={i} />)
    }
    return secs
  }


  squareGrid = () => {
    const { mode } = this.props
    const width = window.innerWidth - 250
    const height = window.innerHeight - 50

    let vert = [], horz = [], refs = [], diag = [];

    for (let i = 130; i <= height + 130; i += 130) {
      vert.push(<HorizontalLine key={i + 'HorizontalLine'} top={i} />)
    }

    for (let i = 130; i <= width + 130; i += 130) {
      horz.push(<VerticalLine key={i + 'VerticalLine'} top={i} />)
    }

    for (let y = 130; y <= height + 130; y += 2 * 130) {
      for (let x = 130; x <= width + 2 * 130; x += 2 * 130) {
        refs.push(<ReferencePoint key={`${y}${x}ReferencePoint`} top={y - 4} left={x - 4} />)
        if (mode === 'SquareKaliedo') {
          diag.push(<Diagonal angle={45} key={`${y}${x}Diagonal`} top={y - 130} left={x - 130} />)
          diag.push(<Diagonal angle={135} key={`${y}${x}Diagonal2`} top={y - 130} left={x - 130} />)
        }
      }
    }
    return [...vert, ...horz, ...refs, ...diag]
  }

  triangleGrid = () => {
    const width = window.innerWidth - 250
    const height = window.innerHeight - 50

    let vert = [], refs = [], diag = [];

    for (let i = 112.5; i <= height + 112.5; i += 112.5) {
      vert.push(<HorizontalLine key={i + 'HorizontalLine'} top={i} />)
    }
    let temp = 0;
    for (let y = 0; y <= height + 112.5; y += 112.5) {
      const offset = !(temp % 2) ? 65 : 0;
      for (let x = 0; x <= width + 130; x += 130) {
        diag.push(<Diagonal2 angle={60} key={`${x}${y}Diagonal`} top={y} left={x - offset} />)
        diag.push(<Diagonal2 angle={120} key={`${x}${y}Diagonal23`} top={y} left={x - offset} />)
      }
      temp++;
    }
    let temp2 = 0;
    for (let y = 0; y <= height + 112.5; y += 112.5) {
      const offset = !(temp2 % 2) ? 3 * 130 - 65 : 130;
      for (let x = 0; x <= width + 130; x += 3 * 130) {
        refs.push(<ReferencePoint key={`${y}${x}ReferencePoint`} top={y - 4} left={x + offset - 4}></ReferencePoint>)
      }
      temp2++;
    }
    return [...vert, ...diag, ...refs]
  }

  // used only for the case of hexagon
  hexagonGrid = () => {
    const width = window.innerWidth - 250
    const height = window.innerHeight - 50
    let diag = [];
    let temp = 0;
    for (let y = 0; y <= height + 112.5; y += 112.5) {
      const offset = !(temp % 2) ? 65 : 0;
      for (let x = 130; x <= width + 130; x += 3*130) {
        if(offset){
          diag.push(<Diagonal2 angle={60} key={`${x}${y}Diagonal`} top={y} left={130+x - offset} />)
          diag.push(<Diagonal2 angle={120} key={`${x}${y}Diagonal23`} top={y} left={x - offset} />)
          diag.push(<Diagonal2 angle={0} key={`${x}${y}Diagonal24`} top={y} left={x - offset} />)  
        }else{
          diag.push(<Diagonal2 angle={60} key={`${x}${y}Diagonal`} top={y} left={x - offset-130} />)
          diag.push(<Diagonal2 angle={120} key={`${x}${y}Diagonal23`} top={y} left={130 + x - offset} />)
          diag.push(<Diagonal2 angle={0} key={`${x}${y}Diagonal24`} top={y} left={130 + x - offset} />)  
        }
      }
      temp++;
    }
    return diag
  }


  render() {
    return (
      <Container>
        {this.renderGrid()}
      </Container>
    )
  }

}

export default Index