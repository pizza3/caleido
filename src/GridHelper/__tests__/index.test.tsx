import React from 'react';
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from 'enzyme';
import GridHelper from '../index'


configure({
    adapter: new Adapter()
});

describe('<GridHelper/>', () => {
    it('It shallow renders GridHelper', () => {
        const component = shallow(<GridHelper />)
        expect(toJson(component)).toMatchSnapshot();
    })

    it('It shallow renders GridHelper when props provided', () => {
        const component = shallow(<GridHelper mode='HexagonRotation' />)
        expect(toJson(component)).toMatchSnapshot();
    })
})