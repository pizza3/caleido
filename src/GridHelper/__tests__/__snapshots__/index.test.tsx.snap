// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridHelper/> It shallow renders GridHelper 1`] = `
<styled.div>
  <styled.div
    top={387}
  />
</styled.div>
`;

exports[`<GridHelper/> It shallow renders GridHelper when props provided 1`] = `
<styled.div>
  <styled.div
    key="112.5HorizontalLine"
    top={112.5}
  />
  <styled.div
    key="225HorizontalLine"
    top={225}
  />
  <styled.div
    key="337.5HorizontalLine"
    top={337.5}
  />
  <styled.div
    key="450HorizontalLine"
    top={450}
  />
  <styled.div
    key="562.5HorizontalLine"
    top={562.5}
  />
  <styled.div
    key="675HorizontalLine"
    top={675}
  />
  <styled.div
    key="787.5HorizontalLine"
    top={787.5}
  />
  <styled.div
    angle={60}
    key="00Diagonal"
    left={-65}
    top={0}
  />
  <styled.div
    angle={120}
    key="00Diagonal23"
    left={-65}
    top={0}
  />
  <styled.div
    angle={60}
    key="1300Diagonal"
    left={65}
    top={0}
  />
  <styled.div
    angle={120}
    key="1300Diagonal23"
    left={65}
    top={0}
  />
  <styled.div
    angle={60}
    key="2600Diagonal"
    left={195}
    top={0}
  />
  <styled.div
    angle={120}
    key="2600Diagonal23"
    left={195}
    top={0}
  />
  <styled.div
    angle={60}
    key="3900Diagonal"
    left={325}
    top={0}
  />
  <styled.div
    angle={120}
    key="3900Diagonal23"
    left={325}
    top={0}
  />
  <styled.div
    angle={60}
    key="5200Diagonal"
    left={455}
    top={0}
  />
  <styled.div
    angle={120}
    key="5200Diagonal23"
    left={455}
    top={0}
  />
  <styled.div
    angle={60}
    key="6500Diagonal"
    left={585}
    top={0}
  />
  <styled.div
    angle={120}
    key="6500Diagonal23"
    left={585}
    top={0}
  />
  <styled.div
    angle={60}
    key="7800Diagonal"
    left={715}
    top={0}
  />
  <styled.div
    angle={120}
    key="7800Diagonal23"
    left={715}
    top={0}
  />
  <styled.div
    angle={60}
    key="0112.5Diagonal"
    left={0}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="0112.5Diagonal23"
    left={0}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="130112.5Diagonal"
    left={130}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="130112.5Diagonal23"
    left={130}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="260112.5Diagonal"
    left={260}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="260112.5Diagonal23"
    left={260}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="390112.5Diagonal"
    left={390}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="390112.5Diagonal23"
    left={390}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="520112.5Diagonal"
    left={520}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="520112.5Diagonal23"
    left={520}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="650112.5Diagonal"
    left={650}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="650112.5Diagonal23"
    left={650}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="780112.5Diagonal"
    left={780}
    top={112.5}
  />
  <styled.div
    angle={120}
    key="780112.5Diagonal23"
    left={780}
    top={112.5}
  />
  <styled.div
    angle={60}
    key="0225Diagonal"
    left={-65}
    top={225}
  />
  <styled.div
    angle={120}
    key="0225Diagonal23"
    left={-65}
    top={225}
  />
  <styled.div
    angle={60}
    key="130225Diagonal"
    left={65}
    top={225}
  />
  <styled.div
    angle={120}
    key="130225Diagonal23"
    left={65}
    top={225}
  />
  <styled.div
    angle={60}
    key="260225Diagonal"
    left={195}
    top={225}
  />
  <styled.div
    angle={120}
    key="260225Diagonal23"
    left={195}
    top={225}
  />
  <styled.div
    angle={60}
    key="390225Diagonal"
    left={325}
    top={225}
  />
  <styled.div
    angle={120}
    key="390225Diagonal23"
    left={325}
    top={225}
  />
  <styled.div
    angle={60}
    key="520225Diagonal"
    left={455}
    top={225}
  />
  <styled.div
    angle={120}
    key="520225Diagonal23"
    left={455}
    top={225}
  />
  <styled.div
    angle={60}
    key="650225Diagonal"
    left={585}
    top={225}
  />
  <styled.div
    angle={120}
    key="650225Diagonal23"
    left={585}
    top={225}
  />
  <styled.div
    angle={60}
    key="780225Diagonal"
    left={715}
    top={225}
  />
  <styled.div
    angle={120}
    key="780225Diagonal23"
    left={715}
    top={225}
  />
  <styled.div
    angle={60}
    key="0337.5Diagonal"
    left={0}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="0337.5Diagonal23"
    left={0}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="130337.5Diagonal"
    left={130}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="130337.5Diagonal23"
    left={130}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="260337.5Diagonal"
    left={260}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="260337.5Diagonal23"
    left={260}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="390337.5Diagonal"
    left={390}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="390337.5Diagonal23"
    left={390}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="520337.5Diagonal"
    left={520}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="520337.5Diagonal23"
    left={520}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="650337.5Diagonal"
    left={650}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="650337.5Diagonal23"
    left={650}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="780337.5Diagonal"
    left={780}
    top={337.5}
  />
  <styled.div
    angle={120}
    key="780337.5Diagonal23"
    left={780}
    top={337.5}
  />
  <styled.div
    angle={60}
    key="0450Diagonal"
    left={-65}
    top={450}
  />
  <styled.div
    angle={120}
    key="0450Diagonal23"
    left={-65}
    top={450}
  />
  <styled.div
    angle={60}
    key="130450Diagonal"
    left={65}
    top={450}
  />
  <styled.div
    angle={120}
    key="130450Diagonal23"
    left={65}
    top={450}
  />
  <styled.div
    angle={60}
    key="260450Diagonal"
    left={195}
    top={450}
  />
  <styled.div
    angle={120}
    key="260450Diagonal23"
    left={195}
    top={450}
  />
  <styled.div
    angle={60}
    key="390450Diagonal"
    left={325}
    top={450}
  />
  <styled.div
    angle={120}
    key="390450Diagonal23"
    left={325}
    top={450}
  />
  <styled.div
    angle={60}
    key="520450Diagonal"
    left={455}
    top={450}
  />
  <styled.div
    angle={120}
    key="520450Diagonal23"
    left={455}
    top={450}
  />
  <styled.div
    angle={60}
    key="650450Diagonal"
    left={585}
    top={450}
  />
  <styled.div
    angle={120}
    key="650450Diagonal23"
    left={585}
    top={450}
  />
  <styled.div
    angle={60}
    key="780450Diagonal"
    left={715}
    top={450}
  />
  <styled.div
    angle={120}
    key="780450Diagonal23"
    left={715}
    top={450}
  />
  <styled.div
    angle={60}
    key="0562.5Diagonal"
    left={0}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="0562.5Diagonal23"
    left={0}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="130562.5Diagonal"
    left={130}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="130562.5Diagonal23"
    left={130}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="260562.5Diagonal"
    left={260}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="260562.5Diagonal23"
    left={260}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="390562.5Diagonal"
    left={390}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="390562.5Diagonal23"
    left={390}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="520562.5Diagonal"
    left={520}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="520562.5Diagonal23"
    left={520}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="650562.5Diagonal"
    left={650}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="650562.5Diagonal23"
    left={650}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="780562.5Diagonal"
    left={780}
    top={562.5}
  />
  <styled.div
    angle={120}
    key="780562.5Diagonal23"
    left={780}
    top={562.5}
  />
  <styled.div
    angle={60}
    key="0675Diagonal"
    left={-65}
    top={675}
  />
  <styled.div
    angle={120}
    key="0675Diagonal23"
    left={-65}
    top={675}
  />
  <styled.div
    angle={60}
    key="130675Diagonal"
    left={65}
    top={675}
  />
  <styled.div
    angle={120}
    key="130675Diagonal23"
    left={65}
    top={675}
  />
  <styled.div
    angle={60}
    key="260675Diagonal"
    left={195}
    top={675}
  />
  <styled.div
    angle={120}
    key="260675Diagonal23"
    left={195}
    top={675}
  />
  <styled.div
    angle={60}
    key="390675Diagonal"
    left={325}
    top={675}
  />
  <styled.div
    angle={120}
    key="390675Diagonal23"
    left={325}
    top={675}
  />
  <styled.div
    angle={60}
    key="520675Diagonal"
    left={455}
    top={675}
  />
  <styled.div
    angle={120}
    key="520675Diagonal23"
    left={455}
    top={675}
  />
  <styled.div
    angle={60}
    key="650675Diagonal"
    left={585}
    top={675}
  />
  <styled.div
    angle={120}
    key="650675Diagonal23"
    left={585}
    top={675}
  />
  <styled.div
    angle={60}
    key="780675Diagonal"
    left={715}
    top={675}
  />
  <styled.div
    angle={120}
    key="780675Diagonal23"
    left={715}
    top={675}
  />
  <styled.div
    angle={60}
    key="0787.5Diagonal"
    left={0}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="0787.5Diagonal23"
    left={0}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="130787.5Diagonal"
    left={130}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="130787.5Diagonal23"
    left={130}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="260787.5Diagonal"
    left={260}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="260787.5Diagonal23"
    left={260}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="390787.5Diagonal"
    left={390}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="390787.5Diagonal23"
    left={390}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="520787.5Diagonal"
    left={520}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="520787.5Diagonal23"
    left={520}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="650787.5Diagonal"
    left={650}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="650787.5Diagonal23"
    left={650}
    top={787.5}
  />
  <styled.div
    angle={60}
    key="780787.5Diagonal"
    left={780}
    top={787.5}
  />
  <styled.div
    angle={120}
    key="780787.5Diagonal23"
    left={780}
    top={787.5}
  />
  <styled.div
    key="00ReferencePoint"
    left={321}
    top={-4}
  />
  <styled.div
    key="0390ReferencePoint"
    left={711}
    top={-4}
  />
  <styled.div
    key="0780ReferencePoint"
    left={1101}
    top={-4}
  />
  <styled.div
    key="112.50ReferencePoint"
    left={126}
    top={108.5}
  />
  <styled.div
    key="112.5390ReferencePoint"
    left={516}
    top={108.5}
  />
  <styled.div
    key="112.5780ReferencePoint"
    left={906}
    top={108.5}
  />
  <styled.div
    key="2250ReferencePoint"
    left={321}
    top={221}
  />
  <styled.div
    key="225390ReferencePoint"
    left={711}
    top={221}
  />
  <styled.div
    key="225780ReferencePoint"
    left={1101}
    top={221}
  />
  <styled.div
    key="337.50ReferencePoint"
    left={126}
    top={333.5}
  />
  <styled.div
    key="337.5390ReferencePoint"
    left={516}
    top={333.5}
  />
  <styled.div
    key="337.5780ReferencePoint"
    left={906}
    top={333.5}
  />
  <styled.div
    key="4500ReferencePoint"
    left={321}
    top={446}
  />
  <styled.div
    key="450390ReferencePoint"
    left={711}
    top={446}
  />
  <styled.div
    key="450780ReferencePoint"
    left={1101}
    top={446}
  />
  <styled.div
    key="562.50ReferencePoint"
    left={126}
    top={558.5}
  />
  <styled.div
    key="562.5390ReferencePoint"
    left={516}
    top={558.5}
  />
  <styled.div
    key="562.5780ReferencePoint"
    left={906}
    top={558.5}
  />
  <styled.div
    key="6750ReferencePoint"
    left={321}
    top={671}
  />
  <styled.div
    key="675390ReferencePoint"
    left={711}
    top={671}
  />
  <styled.div
    key="675780ReferencePoint"
    left={1101}
    top={671}
  />
  <styled.div
    key="787.50ReferencePoint"
    left={126}
    top={783.5}
  />
  <styled.div
    key="787.5390ReferencePoint"
    left={516}
    top={783.5}
  />
  <styled.div
    key="787.5780ReferencePoint"
    left={906}
    top={783.5}
  />
</styled.div>
`;
